routers/web/repo/helper.go

```diff
@@ -20,3 +23,22 @@ func MakeSelfOnTop(doer *user.User, users []*user.User) []*user.User {
 	}
 	return users
 }
+
+func HandleGitError(ctx *context.Context, msg string, err error) {
+	if git.IsErrNotExist(err) {
+		refType := ""
+		switch {
+		case ctx.Repo.IsViewBranch:
+			refType = "branch"
+		case ctx.Repo.IsViewTag:
+			refType = "tag"
+		case ctx.Repo.IsViewCommit:
+			refType = "commit"
+		}
+		ctx.Data["NotFoundPrompt"] = ctx.Locale.Tr("repo.tree_path_not_found_" + refType, ctx.Repo.TreePath, url.PathEscape(ctx.Repo.RefName))
```

Comment:
@delvh 
There's a risk when refType is empty? I'm not sure what will happen when the key word not exists.

Reply:
But I don't even know how a user would trigger something like that.
What other ref types are there except branches, commits, and tags?